flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -3861,6 +3862,14 @@ class FIRToLLVMLowering
                                                std::move(pattern)))) {
       signalPassFailure();
     }
+
+    // Run pass to add comdats to functions that have weak linkage on relevant platforms
+    if (fir::getTargetTriple(mod).supportsCOMDAT()) {
```

Comment:
Would it be easier to move this if statement to `flang/include/flang/Tools/CLOptions.inc` and re-use the existing pass manager? To me it feels a bit weird to create a new pass manager inside of a pass.

Reply:
I did it here mostly because conceptually it's the place in the pipeline that it needs to happen; Another pass pipeline is actually run inside this pass just above this (`mlir::OpPassManager mathConvertionPM("builtin.module");`) so I thought it might not be that odd to do. 
I guess I could add it around here somewhere instead to use the same original PM, but do we have enough info to get the target triple info there? https://github.com/llvm/llvm-project/blob/32db121b29f78e4c41116b2a8f1c730f9522b202/flang/include/flang/Tools/CLOptions.inc#L161